### PR TITLE
Update Error prone 2.4

### PR DESCRIPTION
**Pull request by nicerloop**
Update error-prone to 2.4.0 using ErrorProneJavaCompiler implementation of java.tools.JavaCompiler based on plexus-compiler-javac

### CODE REVIEW

**Comment by pzygielo** [comment]
This would solve #77, wouldn't it?

**Comment by nicerloop** [comment]
Yes. I just had a look at the failing builds : mac builds fail on j2objc missing , need activate profile to skip tests; jdk14 tests fail on additional warnings since jdk11

**Comment by olamy** [comment]
@nicerloop ah yes sorry for the failing ghactions, I just added that yesterday and obviously some are failing for bad reasons.

File: plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java [hunk 1]
```diff
@@ -43,31 +40,69 @@
...
+    }
+    
+    private ClassLoader originalTCCL;
+    
     public void setUp()
         throws Exception
     {
         super.setUp();
-        setForceJavacCompilerUse( true );
+        if ( java8() ) {
```
Comment (olamy) [inline plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java:53]: does it mean we do not test with new jdk such 14?

Comment (nicerloop) [reply plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java:53]: I did test on LTS jdk8 and jdk11, and warning count changes ; I did not test with other jdk

Comment (nicerloop) [reply plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java:53]: I did test on LTS jdk8 and jdk11, and warning count changes ; I did not test with other jdk

Comment (olamy) [reply plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java:53]: I disable the `j2objc` test for ghactions.
Still some issues with jdk14

Comment (olamy) [reply plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java:53]: @nicerloop rebase with master I have disabled jdk14 until we get this fixed

Comment (nicerloop) [reply plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java:53]: I locally have issues with jdk14 in plexus-compiler-eclipse("The type java.lang.Object cannot be resolved") and plexus-compiler-javac (source/target 1.6 not supported anymore, use 7 or later)

Comment (nicerloop) [reply plexus-compilers/plexus-compiler-javac-errorprone/src/test/java/org/codehaus/plexus/compiler/javac/JavacErrorProneCompilerTest.java:53]: It seems to me master does not build with jdk14

**Comment by nicerloop** [comment]
Rebased on master, all checks OK.

**Comment by nicerloop** [comment]
Rebased on master

**Pull request merged**